src/main/java/net/dv8tion/jda/internal/entities/GuildImpl.java

```diff
@@ -1427,75 +1394,81 @@ private AuditableRestAction<Void> timeoutUntilById0(@Nonnull String userId, @Nul
 
     @Nonnull
     @Override
-    public AuditableRestAction<Void> deafen(@Nonnull Member member, boolean deafen)
+    public AuditableRestAction<Void> deafen(@Nonnull UserSnowflake user, boolean deafen)
     {
-        Checks.notNull(member, "Member");
-        checkGuild(member.getGuild(), "Member");
+        Checks.notNull(user, "User");
         checkPermission(Permission.VOICE_DEAF_OTHERS);
 
-        GuildVoiceState voiceState = member.getVoiceState();
-        if (voiceState != null)
+        Member member = resolveMember(user);
+        if (member != null)
         {
-            if (voiceState.getChannel() == null)
-                throw new IllegalStateException("Can only deafen members who are currently in a voice channel");
-            if (voiceState.isGuildDeafened() == deafen)
-                return new CompletedRestAction<>(getJDA(), null);
+            checkGuild(member.getGuild(), "Member");
```

Comment:
Technically this checkGuild isn't needed anymore because we return `null` from `resolveMember` if the member isn't in this guild.